File: android/src/main/java/top/huic/tencent_im_plugin/message/ImageMessageNode.java
package top.huic.tencent_im_plugin.message;

import com.tencent.imsdk.v2.V2TIMImageElem;
import com.tencent.imsdk.v2.V2TIMManager;
import com.tencent.imsdk.v2.V2TIMMessage;

import top.huic.tencent_im_plugin.message.entity.ImageMessageEntity;

/**
 * 图片消息节点
 */
public class ImageMessageNode extends AbstractMessageNode<V2TIMImageElem, ImageMessageEntity> {
    @Override
    public V2TIMMessage getV2TIMMessage(ImageMessageEntity entity) {
        return V2TIMManager.getMessageManager().createImageMessage(entity.getPath());
    }

    @Override
    public String getNote(V2TIMImageElem elem) {
        return "[图片]";
    }

    @Override
    public ImageMessageEntity analysis(V2TIMImageElem elem) {
        ImageMessageEntity entity = new ImageMessageEntity();
        entity.setPath(elem.getPath());
        entity.setImageData(elem.getImageList());
        return entity;
    }

    @Override
    public Class<ImageMessageEntity> getEntityClass() {
        return ImageMessageEntity.class;
    }
}

File: ios/Classes/message/ImageMessageNode.swift
import ImSDK

//
//  TextMessageNode.swift
//  tencent_im_plugin
//
//  Created by 蒋具宏 on 2020/3/13.
//  图片消息节点
public class ImageMessageNode: AbstractMessageNode {

    override func getV2TIMMessage(params: [String: Any]) -> V2TIMMessage {
        let path: String = getParam(params: params, paramKey: "path")!;
        return V2TIMManager.sharedInstance().createImageMessage(path)
    }

    override func getNote(elem: V2TIMElem) -> String {
        "[图片]";
    }

    override func analysis(elem: V2TIMElem) -> AbstractMessageEntity {
        ImageMessageEntity(elem: elem as! V2TIMImageElem)
    }
}
